docs/documents/query.mdx

```diff
@@ -1,53 +1,125 @@
 ---
 title: "Query documents"
-description: "Use the `query` method to search for documents."
+description: "Use the `query` function to search for documents."
 ---
 
 import Support from "/snippets/support.mdx";
 
-TopK allows you to query documents using a SQL-inspired syntax. It comes with **semantic search**, **text search**, **vector search**, and **filtering capabilities** out of the box.
+TopK provides a SQL-inspired syntax for querying documents. It features built-in **semantic search**, **text search**, **vector search**, metadata **filtering** as well as **reranking** capabilities.
 
-TopK's declarative query builder allows you to select fields, chain filters, apply vector/text search in a composable way.
+With TopK's declarative query builder, you can easily select fields, chain filters, and apply vector/text search in a composable manner.
 
-## Semantic search
+## Query structure
 
-To perform a semantic search on your documents, use the `semantic_similarity` function. Query you want to search for is automatically embedded using the `model` specified in the `semantic_index` definition.
+In TopK, a query consists of multiple stages:
+
+* **Select stage** - Select static or computed fields that will be returned in the query results
+  * these fields can be used in stages such as Filter, TopK or Rerank
+* **Filter stage** - Filter the documents that will be returned in the query results
+  * filters can be applied to static fields, computed fields such as `vector_distance()` or `semantic_similarity()` or custom properties computed inside `select()`
+* **TopK stage** - Return the top `k` results based on the provided logical expression
+* **Count stage** - Return the total number of documents matching the query
+* **Rerank stage** - Rerank the results
+
+All queries **must** have either TopK or Count collection stage.
+
+<Tip>
+  You can stack multiple select, filter, topk and rerank stages in a single query.
+</Tip>
+
+A typical query in TopK looks as follows:
+
+```mermaid
+flowchart TD
+    A[Select]
+    A --> B[Filter]
+    B --> C[TopK]
+    B --> E[Count]
+    C -.-> D[Rerank]
+    D --> G@{ shape: stadium, label: "Results collection" }
+    E --> H@{ shape: stadium, label: "Results count" }
+```
+
+
+## Select
+
+The `select()` function is used to initialize the select stage of a query. It accepts a key-value pair of field names and field expressions:
 
 <CodeGroup>
 
 ```python Python
-from topk_sdk.query import select, fn, match
+from topk_sdk.query import select, field
+
+client.collection("books").query(
+  select(
+    title=field("title"),
+    "published_year" # you can also select the static fields directly
+  )
+  ...
+)
+```
+
+```js Javascript
+import { select, field } from "topk-js/query";
+
+await client.collection("books").query(
+  select({
+    title: field("title"),
+  })
+  ...
+)
+```
+
+</CodeGroup>
+
+### Select expressions
```

Comment:
this is the same as the example above?

Reply:
probably a mistake and should show `fn.` usage

Reply:
The goal was to provide docs for both `select()`(query builder) as well as `field()` - select expression.
But I see your point that both examples look almost identical. Perhaps I should not use `field()` expr inside the `select()` docs